projects/schematics/src/migrations/4_0/constructor-deprecations/data/search-box.component.migration.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  ROUTING_SERVICE,
+  SEARCH_BOX_COMPONENT,
+  SPARTACUS_CORE,
+  SPARTACUS_STOREFRONTLIB,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const SEARCH_BOX_COMPONENT_MIGRATION: ConstructorDeprecation = {
+  // storefrontlibs\scr\cms-components\navigation\search-box\search-box.component.ts
+  class: SEARCH_BOX_COMPONENT,
+  importPath: SPARTACUS_STOREFRONTLIB,
+  deprecatedParams: [],
```

Comment:
In deprecated params you must list all the parameters of previous constructor you want to change.
In this case: SearchBoxComponentService, CmsComponentData and WindowRef